django/db/models/fields/related_objects.py

```diff
@@ -0,0 +1,315 @@
+"""
+"Rel objects" for related fields.
+
+"Rel objects" (for lack of a better name) carry information about the relation
+modeled by a related field and provide some utility functions. They're stored
+in the ``rel`` attribute of the field.
```

Comment:
One annoying bit is that we'll now have a `ReverseManyToManyField` while technically one shouldn't be needed, but I guess that's something we can address in the future.

Dunno about size of `related_fields.py`, maybe `related_fields.py` + `related_reverse_fields.py`? Most other Django fields fit into a single large file (`__init__.py`) without it being a problem, as I see it the problem with `related.py` was more about each field having a lot of moving pieces, and some of them being terribly misnamed.